### PR TITLE
docs: API consumption

### PR DESCRIPTION
**Pull request by shimks**
- Write docs for using Swagger UI to test your REST API endpoints
- Part of Thinking in Loopback/Best practices
- Connected to https://github.com/strongloop/loopback-next/issues/548

### CODE REVIEW

File: pages/en/lb4/Preparing-the-API-for-consumption.md [hunk 1]
```diff
@@ -14,11 +14,23 @@ This article continues off from [Implmenting features](./Implementing-features.
 
 ## Preparing your API for consumption
 
-{% include content/tbd.html %}
+### Interacting with you API
 
-- Documentation
-- Public endpoint for the swagger spec (for discovery)
-- more?
+When your application is run, its documentation is generated into `swagger.json` file and can be accessed from your server through `/swagger.json`. You can also interact with the API endpoints defined in `swagger.json` file through Swagger UI (`/swagger-ui`), which allows users to directly interact with the documented API endpoints of the application.
```
Comment (b-admike) [inline pages/en/lb4/Preparing-the-API-for-consumption.md]: You can also mention that we provide the OpenAPI spec as well and in both yaml and json flavours for both versions.

Comment (kjdelisle) [reply pages/en/lb4/Preparing-the-API-for-consumption.md]: I wouldn't yet do that; we haven't confirmed OpenAPI support.

Until we've finalized our plans, you can leave that part out.

File: pages/en/lb4/Preparing-the-API-for-consumption.md [hunk 2]
```diff
@@ -14,11 +14,23 @@ This article continues off from [Implmenting features](./Implementing-features.
 
 ## Preparing your API for consumption
 
-{% include content/tbd.html %}
+### Interacting with you API
```
Comment (kjdelisle) [inline pages/en/lb4/Preparing-the-API-for-consumption.md]: `Interacting with your API`

You're missing the "r" on "your"

File: pages/en/lb4/Preparing-the-API-for-consumption.md [hunk 3]
```diff
@@ -14,11 +14,23 @@ This article continues off from [Implmenting features](./Implementing-features.
...
 
-- Documentation
-- Public endpoint for the swagger spec (for discovery)
-- more?
+When your application is run, its documentation is generated into `swagger.json` file and can be accessed from your server through `/swagger.json`. You can also interact with the API endpoints defined in `swagger.json` file through Swagger UI (`/swagger-ui`), which allows users to directly interact with the documented API endpoints of the application.
+
+We'll use the repo [loopback4-example-getting-started](https://github.com/strongloop/loopback4-example-getting-started) to demonstrate how Swagger UI can be used to test your endpoints.
+
```
Comment (jannyHou) [inline pages/en/lb4/Preparing-the-API-for-consumption.md]: nitpick feel free to ignore: I think adding something to hint users run the app here would be better, like 
- `git clone https://github.com/strongloop/loopback4-example-getting-started`
- `npm i`
- `npm start`

Or just "run the app"

**Review by jannyHou — APPROVED** [review]

**Review by bschrammIBM — CHANGES_REQUESTED** [review]
I have edited this content and suggest the following edits:

**Comment by bschrammIBM** [comment]
## Preparing your API for consumption

 -{% include content/tbd.html %}		 +### Interacting with your API

 -- Documentation		 +After your application has been run, its documentation is generated into a `swagger.json` file which can be accessed from your server through `/swagger.json`. Your spec can also be accessed in YAML flavour through the `/swagger.yaml` endpoint. The API endpoints defined in your Swagger spec can be accessed from the Swagger UI (at `/swagger-ui`), which allows users to directly interact with the documented API endpoints of the application.
 -- Public endpoint for the swagger spec (for discovery)		 +
 -- more?		 +We'll use the repo [loopback4-example-getting-started](https://github.com/strongloop/loopback4-example-getting-started) to demonstrate how Swagger UI can be used to test your endpoints.
 +
 +First, git clone the repository and install its dependencies:
 +
 +```
 +$ git clone https://github.com/strongloop/loopback4-example-getting-started
 +$ cd loopback4-example-getting-started
 +$ npm i
 +$ npm start
 +```
 +
 +Open http://localhost:3000/swagger-ui to see your API endpoints.
 +
 +{% include image.html file="lb4/10000000.png" alt="" %}
 +
 +The Swagger UI displays all of the endpoints defined in your application.
 +
 +{% include image.html file="lb4/10000001.png" alt="" %}
 +
 +Clicking on one of the endpoints will show the endpoint's documentation as defined in your API specification. Next, click on `Try It Out` to send a request to the endpoint. If the endpoint takes parameters, assign the values before the request is sent. If the parameter involves a body, a template is given for you to edit as specified in your specification. Choose `Execute` to send the request:
 +
 +{% include image.html file="lb4/10000002.png" alt="" %}
 +
 +The response to the request can be seen below the `Execute` button, where the response code and the body are displayed. Ideally, each endpoint should be tested with good and bad inputs to confirm that the responses are as expected.

File: pages/en/lb4/Preparing-the-API-for-consumption.md [hunk 4]
```diff
@@ -14,11 +14,37 @@ This article continues off from [Implmenting features](./Implementing-features.
...
+The Swagger UI displays all of the endpoints defined in your application.
+
+{% include image.html file="lb4/10000001.png" alt="" %}
+
+Clicking on one of the endpoints will show the endpoint's documentation as defined in your API spec. Next, click on `Try It Out` to send a request to the endpoint. If the endpoint takes parameters, assign the values before the request is sent. If the parameter involves a body, a template is given for you to edit as specified in your spec. Click `Execute` to send the request:
+
+{% include image.html file="lb4/10000002.png" alt="" %}
+
+The response to the request can be seen below the `Execute` button, where the response code and the body are displayed. Ideally, each endpoint should be tested with good and bad inputs to confirm that the given responses are as expected.
```
Comment (b-admike) [inline pages/en/lb4/Preparing-the-API-for-consumption.md]: I'd say `returned responses` or just `responses` here instead of `given responses`

**Review by b-admike — APPROVED** [review]

**Comment by kjdelisle** [comment]
@bschrammIBM Would you be able to go through the individual files and provide your recommendations on the areas in question? If you go into [this section](https://github.com/strongloop/loopback.io/pull/536/files) and click on the lines of text, you can leave comments on the areas of concern.

This will make it easier for @shimks to apply feedback, and for others to leave comments in response (it might be that we don't agree with a proposed change and would like to respond in-line).

**Review by bschrammIBM — APPROVED** [review]
this version is approved

**Pull request merged**